### PR TITLE
Optimize http_bench_json_ops example

### PR DESCRIPTION
**Pull request by littledivy**

### CODE REVIEW

**Comment by ry** [comment]
Let’s publish polloi as a crate?

**Review by ry — COMMENTED** [review]
It seems odd to have this one benchmark use polloi while the rest of the system uses tokio...

**Comment by piscisaureus** [comment]
I don't think it is working. The last benchmark run on CI run gets [1 request per second](https://github.com/denoland/deno/actions/runs/3368777169/jobs/5587702432#step:42:1033).

```
http_benchmark testing CORE http_bench_json_ops
/home/runner/work/deno/deno/target/release/examples/http_bench_json_ops
http_bench_ops listening on http://127.0.0.1:4570/
/home/runner/work/deno/deno/third_party/prebuilt/linux64/wrk -d 10s --latency http://127.0.0.1:4570/
Running 10s test @ http://127.0.0.1:4570/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   435.40us  213.60us 767.00us   60.00%
    Req/Sec    50.00      0.00    50.00    100.00%
  Latency Distribution
     50%  507.00us
     75%  549.00us
     90%  767.00us
     99%  767.00us
  10 requests in 10.02s, 510.00B read
Requests/sec:      1.00
Transfer/sec:      50.89B
```

File: core/examples/http_bench_json_ops/http_bench_json_ops.js [hunk 1]
```diff
@@ -13,27 +15,24 @@ const responseBuf = new Uint8Array(
 
 /** Listens on 0.0.0.0:4570, returns rid. */
 function listen() {
-  return Deno.core.ops.op_listen();
+  return ops.op_listen();
```
Comment (sno2) [inline core/examples/http_bench_json_ops/http_bench_json_ops.js:18]: Isn't this opening up users to prototype vulnerabilities if they follow the example then?

```
$ deno eval "Deno.core.ops.op_listen = 23;console.log(Deno.core.ops.op_listen);"
23
```

**Review by ry — APPROVED** [review]
LGTM

File: core/Cargo.toml [hunk 2]
```diff
@@ -40,9 +40,10 @@ v8.workspace = true
 
 [[example]]
 name = "http_bench_json_ops"
-path = "examples/http_bench_json_ops.rs"
+path = "examples/http_bench_json_ops/main.rs"
 
 # These dependencies are only used for the 'http_bench_*_ops' examples.
 [dev-dependencies]
 deno_ast.workspace = true
+mio = { version = "0.8", features = ["os-poll", "net"] }
```
Comment (ry) [inline core/Cargo.toml:48]: Remove

File: core/examples/http_bench_json_ops/http_bench_json_ops.js [hunk 3]
```diff
@@ -13,27 +15,29 @@ const responseBuf = new Uint8Array(
...
-    }
-  } catch (e) {
-    if (
-      !e.message.includes("Broken pipe") &&
-      !e.message.includes("Connection reset by peer")
-    ) {
-      throw e;
+      await read(rid, requestBuf);
+      if (!ops.op_try_write(rid, responseBuf)) {
+        console.log("write failed");
```
Comment (bartlomieju) [inline core/examples/http_bench_json_ops/http_bench_json_ops.js:35]: Should this be removed?

File: core/examples/http_bench_json_ops/main.rs [hunk 4]
```diff
@@ -67,25 +45,19 @@ impl TryFrom<std::net::TcpListener> for TcpListener {
...
   }
 }
 
 struct TcpStream {
   rd: AsyncRefCell<tokio::net::tcp::OwnedReadHalf>,
   wr: AsyncRefCell<tokio::net::tcp::OwnedWriteHalf>,
-  // When a `TcpStream` resource is closed, all pending 'read' ops are
-  // canceled, while 'write' ops are allowed to complete. Therefore only
-  // 'read' futures are attached to this cancel handle.
-  cancel: CancelHandle,
```
Comment (bartlomieju) [inline core/examples/http_bench_json_ops/main.rs:81]: Why is this removed?

File: core/examples/http_bench_json_ops/main.rs [hunk 5]
```diff
@@ -94,15 +66,21 @@ impl TcpStream {
...
+  }
 }
 
 impl Resource for TcpStream {
   deno_core::impl_readable_byob!();
   deno_core::impl_writable!();
 
-  fn close(self: Rc<Self>) {
-    self.cancel.cancel()
-  }
```
Comment (bartlomieju) [inline core/examples/http_bench_json_ops/main.rs:105]: Ditto

File: core/examples/http_bench_json_ops/main.rs (hunk 4 above)
Comment (littledivy) [reply core/examples/http_bench_json_ops/main.rs:81]: Because cancellation is not used in the benchmark.

File: core/examples/http_bench_json_ops/http_bench_json_ops.js (hunk 3 above)
Comment (littledivy) [reply core/examples/http_bench_json_ops/http_bench_json_ops.js]: oh yes nice catch

**Review by ry — APPROVED** [review]
LGTM
